modules/library/main/src/main/java/org/geotools/data/DataUtilities.java

```diff
@@ -793,7 +793,7 @@ public static SimpleFeature template(SimpleFeatureType featureType, String featu
      * @throws ArrayIndexOutOfBoundsException If the number of provided values does not match the
      *     featureType
      */
-    public static SimpleFeature template(SimpleFeatureType featureType, Object[] providedValues) {
```

Comment:
Dangerous conversion, will trip client code that is not converted to not using arrays.

Reply:
Will it? It is too bad as this is one case where it may be a usability improvement ...

Reply:
That's what I understood reading articles around, but then tried in a debugger, and the array was not provided as an item, but as the full argument.
The case on which I based the above comment is "Array.asList", which did bite me in the past, you think you're passing in an array to get a list, and get a list with a single value, the array itself. Looking again at the issue, it seems to be happening only if the users of the method tried to pass arrays of primitives:
https://javadevnotes.com/java-array-to-list-examples